src/main/java/com/ontology/service/impl/ToolsServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+package com.ontology.service.impl;
+
+import com.alibaba.fastjson.JSON;
+import com.alibaba.fastjson.JSONArray;
+import com.alibaba.fastjson.JSONObject;
+import com.github.ontio.account.Account;
+import com.github.ontio.sdk.manager.ECIES;
+import com.ontology.controller.vo.OrderListResp;
+import com.ontology.dao.OntId;
+import com.ontology.dao.Order;
+import com.ontology.dao.OrderData;
+import com.ontology.exception.OntIdException;
+import com.ontology.mapper.OntIdMapper;
+import com.ontology.mapper.OrderDataMapper;
+import com.ontology.mapper.OrderMapper;
+import com.ontology.model.Result;
+import com.ontology.secure.SecureConfig;
+import com.ontology.utils.ErrorInfo;
+import com.ontology.utils.Helper;
+import com.ontology.utils.SDKUtil;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.*;
+
+@Slf4j
+@Service
+@Transactional(rollbackFor = Exception.class)
+public class ToolsServiceImpl implements ToolsService {
+    @Autowired
+    private SDKUtil sdk;
+    @Autowired
+    private OntIdMapper ontIdMapper;
+    @Autowired
+    private OrderMapper orderMapper;
+    @Autowired
+    private SecureConfig secureConfig;
+    @Autowired
+    private OrderDataMapper orderDataMapper;
+
+    @Override
+    public List<OrderListResp> queryList(String action, Integer provider, String buyerOntid) {
+        String queryType = null;
+        if (0==provider) {
+            // 需求方
+            queryType = "buyer_ontid";
+        } else if (1==provider) {
+            // 提供方
+            queryType = "seller_ontid";
+        } else {
+            throw new OntIdException(action, ErrorInfo.PARAM_ERROR.descCN(), ErrorInfo.PARAM_ERROR.descEN(), ErrorInfo.PARAM_ERROR.code());
+        }
+        List<Order> orderList = orderMapper.getOrderList(queryType,buyerOntid);
```

Comment:
Maybe you should change `buyerOntid` to `ontid`. What I understand in here is both `buyer` and `seller` can use this interface.

Reply:
ok